src/version.c

```diff
@@ -41,6 +41,31 @@ stumpless_get_version( void ) {
   return version;
 }
 
+int
+stumpless_version_cmp ( const struct stumpless_version * version_x, 
+                         const struct stumpless_version * version_y) {
+  if ( version_x == NULL || version_y == NULL ) {
+    return 666;
```

Comment:
before returning the error code, be sure to call `raise_argument_empty` to set the library error code appropriately first. 

I apologize once again: this is something I haven't documented well and I'll add a section to the development guide about it. Until I get some time to do that, check out the `src/param.c` file for lots of examples for doing this to signify that an argument was empty. The short version is that all functions (except for destructors and error handling code itself) raise errors using the `raise_` family of internal functions in `include/private/error.h` when they need to.

Reply:
Thanks for your detailed reply! I'll get back to you after I've done all these changes.